### PR TITLE
fix reading newly written encrypted files before their cache entry is written

### PR DESCRIPTION
**Pull request by icewind1991**
Keep track of which files we know are encrypted so we know to decrypt them even if they are not in the cache.

This fixes an interaction with `files_accesscontrol` where it tried to check the mimetype before the item was added to cache, resulting in it always detecting the mimetype as `application/octet-stream` (note that in a later step the mimetype was detected properly)

To test:

- Apply https://github.com/nextcloud/server/pull/34579 to fix another accesscontrol+encryption issue
- Enable server-side encryption and files_accesscontrol
- Setup an access control rule blocking files with the mimetype `application/octet-stream`
- Attempt to upload any file

### CODE REVIEW

**Review by CarlSchwan — APPROVED** [review]

File: lib/private/Files/Storage/Wrapper/Encryption.php [hunk 1]
```diff
@@ -973,8 +979,13 @@ protected function getHeader($path) {
 		// first check if it is an encrypted file at all
 		// We would do query to filecache only if we know that entry in filecache exists
 
-		$info = $this->getCache()->get($path);
-		if (isset($info['encrypted']) && $info['encrypted'] === true) {
+		$isEncrypted = $this->encryptedPaths->get($realFile);
```
Comment (PVince81) [inline lib/private/Files/Storage/Wrapper/Encryption.php:982]: should the comment above be adjusted ?

**Review by PVince81 — APPROVED** [review]
:+1: see minor comment

**Comment by PVince81** [comment]
encryption tests are failing

**Comment by PVince81** [comment]
sounds a bit similar to #25768

**Pull request merged**